crates/sui-core/src/authority/authority_store.rs

```diff
@@ -62,52 +69,95 @@ pub struct SuiDataStore<const ALL_OBJ_VER: bool, const USE_LOCKS: bool, S> {
     /// composite key of the SuiAddress of their owner and the object ID of the object.
     /// This composite index allows an efficient iterator to list all objected currently owned
     /// by a specific user, and their object reference.
+    ///
+    /// Consistency requirements:
+    /// - Only exists to satisfy read queries from clients, and should be moved to IndexStore
+    /// - Eventual consistency with respect to certificate processing - the owner index must
+    ///   eventually be updated after a certificate is processed.
+    /// - Read-after-write consistency can be provided if required, such that if client has a
+    ///   signed effects cert they cannot be served the old owner_index entry.
     owner_index: DBMap<(SuiAddress, ObjectID), ObjectRef>,
 
     /// This is map between the transaction digest and transactions found in the `transaction_lock`.
     /// NOTE: after a lock is deleted (after a certificate is processed) the corresponding entry here
     /// could be deleted, but right now this is only done on gateways, not done on authorities.
+    ///
+    /// Consistency requirements:
+    /// - Must never contain an entry for a TransactionDigest unless locks have been succesfully
+    ///   acquired for that tx.
+    /// - Read-after-write consistency for the client: If the Validator::transaction RPC returns
+    ///   successfully, this map must be up to date. If the RPC fails before returning, the client
+    ///   retries, which will eventually cause this map to be up to date.
     transactions: DBMap<TransactionDigest, TransactionEnvelope<S>>,
 
     /// This is a map between the transaction digest and the corresponding certificate for all
     /// certificates that have been successfully processed by this authority. This set of certificates
     /// along with the genesis allows the reconstruction of all other state, and a full sync to this
     /// authority.
+    ///
+    /// Consistency requirements:
+    /// - Read-after-write with respect to Validator::confirmation_transaction
+    /// - A certificate must be added to this map *before* the certificate's input object
```

Comment:
I agree with @velvia that these strong properties are incidental to the current design:
* the requirement is that once we set new locks eventually objects should be created and the certificate registered. But this could well be eventual indeed.
* However in our current design quite a few subsystems would assume that if we have executed the cert others and internal systems should be able to request it without errors. (also assumed in sync code in aggregator / gateway).